types/block_test.go

```diff
@@ -113,6 +113,44 @@ func TestBlockID(t *testing.T) {
 	}
 }
 
+// TestHeaderID probes the ID function of the BlockHeader type.
+func TestHeaderID(t *testing.T) {
```

Comment:
Where would be a good place for the sanity check? This test does verify that `BlockHeader.ID()` and `Block.ID()` match for the test blocks FWIW.